### ISSUE TITLE
Problems with Content-Type: application/json in 1.1.0

### ISSUE THREAD

**Issue by pdobacz** [fanduel/jsonrpc2-elixir#14]
Hi, I think I am getting a similar problem to https://github.com/fanduel/jsonrpc2-elixir/issues/8?

I am not sure but I think this still doesn't work if I run JSONRPC2 in a Supervisor. Here's the way to reproduce:

mix.exs:
```
defmodule JsonTest.MixProject do
  use Mix.Project

  def project do
    [
      app: :json_test,
      version: "0.1.0",
      elixir: "~> 1.6",
      start_permanent: Mix.env() == :prod,
      deps: deps()
    ]
  end

  def application do
    [
      extra_applications: [:logger],
      mod: {JsonTest.Application, []}
    ]
  end

  defp deps do
    [
      {:jsonrpc2, "~> 1.1"}, {:poison, "~> 3.1"}, {:plug, "~> 1.3"}, {:cowboy, "~> 1.1"}
    ]
  end
end
```

application.ex:
```
defmodule JsonTest.Application do
  use Application

  def start(_type, _args) do
    children = [
      JSONRPC2.Servers.HTTP.child_spec(:http, JsonTest)
    ]
    opts = [strategy: :one_for_one, name: JsonTest.Supervisor]
    Supervisor.start_link(children, opts)
  end
end
```

json_test.ex:
```
defmodule JsonTest do
  use JSONRPC2.Server.Handler

  def handle_request(method, params) do
    :ok
  end
end
```

Then with JSONRPC2 1.1.0 as above I'm getting:
```
$ curl "localhost:4000/jsonrpc" -H "Content-Type: plain/text" -d '{"jsonrpc":"2.0", "method": "hello", "id": 1}'
{"result":"ok","jsonrpc":"2.0","id":1}
$ curl "localhost:4000/jsonrpc" -H "Content-Type: application/json" -d '{"jsonrpc":"2.0", "method": "hello", "id": 1}'
{"jsonrpc":"2.0","id":null,"error":{"message":"Invalid Request","code":-32600}}
```

If I downgrade to 1.0.3 then all is good:
```
Resolving Hex dependencies...
Dependency resolution completed:
Unchanged:
  cowboy 1.1.2
  cowlib 1.0.2
  mime 1.3.0
  plug 1.6.1
  poison 3.1.0
  ranch 1.3.2
Downgraded:
  jsonrpc2 1.1.0 => 1.0.3
* Updating jsonrpc2 (Hex package)
$ curl "localhost:4000/jsonrpc" -H "Content-Type: application/json" -d '{"jsonrpc":"2.0", "method": "hello", "id": 1}'
{"result":"ok","jsonrpc":"2.0","id":1}
```

**Comment by ericentin** [comment]
Could you add some logging near here to figure out what is being returned as the request body and why on that version? https://github.com/fanduel/jsonrpc2-elixir/blob/master/lib/jsonrpc2/servers/http/plug.ex#L25

**Comment by pdobacz** [comment]
Thank you @ericentin I did some IO.inspects there like this, for reference:

```
  def call(%{method: "POST"} = conn, handler) do
    IO.inspect conn

    req_body =
      cond do <snip> end
      |> IO.inspect

    resp_body =
      case <snip> end
      |> IO.inspect

    conn
    |> Plug.Conn.put_resp_header("content-type", "application/json")
    |> Plug.Conn.resp(200, resp_body)
    |> IO.inspect
  end
```

I run a correct query and then a failing one (first curl, then httpie)

Inspects for the working request (curl / `plain/text`):
```
%Plug.Conn{
  adapter: {Plug.Adapters.Cowboy.Conn, :...},
  assigns: %{},
  before_send: [],
  body_params: %Plug.Conn.Unfetched{aspect: :body_params},
  cookies: %Plug.Conn.Unfetched{aspect: :cookies},
  halted: false,
  host: "localhost",
  method: "POST",
  owner: #PID<0.676.0>,
  params: %Plug.Conn.Unfetched{aspect: :params},
  path_info: [],
  path_params: %{},
  peer: {{127, 0, 0, 1}, 43604},
  port: 9656,
  private: %{},
  query_params: %Plug.Conn.Unfetched{aspect: :query_params},
  query_string: "",
  remote_ip: {127, 0, 0, 1},
  req_cookies: %Plug.Conn.Unfetched{aspect: :cookies},
  req_headers: [
    {"host", "localhost:9656"},
    {"user-agent", "curl/7.47.0"},
    {"accept", "*/*"},
    {"content-length", "75"},
    {"content-type", "application/x-www-form-urlencoded"}
  ],
  request_path: "/",
  resp_body: nil,
  resp_cookies: %{},
  resp_headers: [{"cache-control", "max-age=0, private, must-revalidate"}],
  scheme: :http,
  script_name: [],
  secret_key_base: nil,
  state: :unset,
  status: nil
}


"{\"params\":{\"transaction\": \"\"}, \"method\": \"submit\", \"jsonrpc\": \"2.0\",\"id\":0}"


"{\"jsonrpc\":\"2.0\",\"id\":0,\"error\":{\"message\":\"Internal error\",\"data\":\"malformed_transaction\",\"code\":-32603}}"


%Plug.Conn{
  adapter: {Plug.Adapters.Cowboy.Conn, :...},
  assigns: %{},
  before_send: [],
  body_params: %Plug.Conn.Unfetched{aspect: :body_params},
  cookies: %Plug.Conn.Unfetched{aspect: :cookies},
  halted: false,
  host: "localhost",
  method: "POST",
  owner: #PID<0.676.0>,
  params: %Plug.Conn.Unfetched{aspect: :params},
  path_info: [],
  path_params: %{},
  peer: {{127, 0, 0, 1}, 43604},
  port: 9656,
  private: %{},
  query_params: %Plug.Conn.Unfetched{aspect: :query_params},
  query_string: "",
  remote_ip: {127, 0, 0, 1},
  req_cookies: %Plug.Conn.Unfetched{aspect: :cookies},
  req_headers: [
    {"host", "localhost:9656"},
    {"user-agent", "curl/7.47.0"},
    {"accept", "*/*"},
    {"content-length", "75"},
    {"content-type", "application/x-www-form-urlencoded"}
  ],
  request_path: "/",
  resp_body: "{\"jsonrpc\":\"2.0\",\"id\":0,\"error\":{\"message\":\"Internal error\",\"data\":\"malformed_transaction\",\"code\":-32603}}",
  resp_cookies: %{},
  resp_headers: [
    {"cache-control", "max-age=0, private, must-revalidate"},
    {"content-type", "application/json"}
  ],
  scheme: :http,
  script_name: [],
  secret_key_base: nil,
  state: :set,
  status: 200
}
```

Inspects for the failing request (httpie / `application/json`):

```
%Plug.Conn{
  adapter: {Plug.Adapters.Cowboy.Conn, :...},
  assigns: %{},
  before_send: [],
  body_params: %Plug.Conn.Unfetched{aspect: :body_params},
  cookies: %Plug.Conn.Unfetched{aspect: :cookies},
  halted: false,
  host: "localhost",
  method: "POST",
  owner: #PID<0.677.0>,
  params: %Plug.Conn.Unfetched{aspect: :params},
  path_info: [],
  path_params: %{},
  peer: {{127, 0, 0, 1}, 43606},
  port: 9656,
  private: %{},
  query_params: %Plug.Conn.Unfetched{aspect: :query_params},
  query_string: "",
  remote_ip: {127, 0, 0, 1},
  req_cookies: %Plug.Conn.Unfetched{aspect: :cookies},
  req_headers: [
    {"host", "localhost:9656"},
    {"accept", "application/json, */*"},
    {"user-agent", "HTTPie/0.9.9"},
    {"accept-encoding", "gzip, deflate"},
    {"connection", "keep-alive"},
    {"content-type", "application/json"},
    {"content-length", "58"}
  ],
  request_path: "/",
  resp_body: nil,
  resp_cookies: %{},
  resp_headers: [{"cache-control", "max-age=0, private, must-revalidate"}],
  scheme: :http,
  script_name: [],
  secret_key_base: nil,
  state: :unset,
  status: nil
}


%Plug.Conn.Unfetched{aspect: :params}


"{\"jsonrpc\":\"2.0\",\"id\":null,\"error\":{\"message\":\"Invalid Request\",\"code\":-32600}}"


%Plug.Conn{
  adapter: {Plug.Adapters.Cowboy.Conn, :...},
  assigns: %{},
  before_send: [],
  body_params: %Plug.Conn.Unfetched{aspect: :body_params},
  cookies: %Plug.Conn.Unfetched{aspect: :cookies},
  halted: false,
  host: "localhost",
  method: "POST",
  owner: #PID<0.677.0>,
  params: %Plug.Conn.Unfetched{aspect: :params},
  path_info: [],
  path_params: %{},
  peer: {{127, 0, 0, 1}, 43606},
  port: 9656,
  private: %{},
  query_params: %Plug.Conn.Unfetched{aspect: :query_params},
  query_string: "",
  remote_ip: {127, 0, 0, 1},
  req_cookies: %Plug.Conn.Unfetched{aspect: :cookies},
  req_headers: [
    {"host", "localhost:9656"},
    {"accept", "application/json, */*"},
    {"user-agent", "HTTPie/0.9.9"},
    {"accept-encoding", "gzip, deflate"},
    {"connection", "keep-alive"},
    {"content-type", "application/json"},
    {"content-length", "58"}
  ],
  request_path: "/",
  resp_body: "{\"jsonrpc\":\"2.0\",\"id\":null,\"error\":{\"message\":\"Invalid Request\",\"code\":-32600}}",
  resp_cookies: %{},
  resp_headers: [
    {"cache-control", "max-age=0, private, must-revalidate"},
    {"content-type", "application/json"}
  ],
  scheme: :http,
  script_name: [],
  secret_key_base: nil,
  state: :set,
  status: 200
}
```

The `Plug.Conn.Unfetched{}` looks suspicious in the latter case, being the request body, and it is actually my handler that then produces the "Invalid request" and "32000". Isn't the condition `cond` broken there, since `conn.params != %Plug.Conn.Unfetched{}` and it `== %Plug.Conn.Unfetched{something}`?

**Comment by ericentin** [comment]
Fixes (hopefully) published as v1.1.1 on Hex
